download_action.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+)
+
+type DownloadAction struct {
+	BaseAction `yaml:",inline"`
+	Url        string // URL for downloading
+	Filename   string // File name, overrides the name from URL.
+	Forced     bool   `yaml:"force"` // Forced re-download of URL
+}
+
+// Function for downloading single file object with http(s) protocol
+func DownloadHttpUrl(url, filename string, forcedRewrite bool) error {
+	log.Printf("Download started: '%s' -> '%s'\n", url, filename)
+
+	// TODO: Proxy support?
+
+	// Check the file. Assume we have the file if file already exists.
+	// Ignore check if flag 'forcedRewrite' is set
+	fi, err := os.Stat(filename)
```

Comment:
Don't do this in this simple way until there is a better way to validate things then just the name (e.g. by also saving and verifying the eTag of a http download).  I'd prefer to see this as a later addition after landing the download action